### DISCUSSION TITLE
Configuring Dual-Stack Networking for Nextcloud AIO & Talk on Bare Metal Kubernetes 1.28

### DISCUSSION THREAD

**Discussion by melbruki** [nextcloud/all-in-one#3872] [Questions] [answered]
Hello Community,

My name is Mohamed, and I am working on deploying Nextcloud AIO, including Nextcloud Talk, on a bare metal Kubernetes cluster (version 1.28) using the official Helm chart. I aim to enable dual-stack (IPv4 and IPv6) accessibility for these services to provide comprehensive network connectivity for our users.

### Current Environment
- **Deployment**: Nextcloud AIO and Nextcloud Talk via Helm Chart
- **Kubernetes Version**: 1.28
- **Infrastructure**: Bare Metal Kubernetes Setup

### Assistance Required
1. **Dual-Stack Configuration**: What specific steps or configurations are needed in the Nextcloud AIO Helm chart to implement dual-stack (IPv4/IPv6) support in a bare metal Kubernetes environment?
2. **Kubernetes Service and Ingress**: How should the Kubernetes Services or Ingress resources be adjusted to efficiently route both IPv4 and IPv6 traffic to Nextcloud AIO and Talk?
3. **Load Balancer and Ingress Controller**: Insights on setting up a Load Balancer or Ingress Controller for dual-stack traffic management in Kubernetes 1.28 on bare metal would be highly beneficial.
4. **Challenges and Compatibility Issues**: Are there known challenges or specific compatibility concerns when running Nextcloud AIO and Talk in a dual-stack setup on bare metal Kubernetes? If so, how can these be mitigated?
5. **Recommendations and Best Practices**: Any best practices, recommendations, or resources for ensuring the seamless operation of Nextcloud AIO and Talk in this dual-stack configuration would be greatly appreciated.

I look forward to your valuable input and thank you in advance for helping enhance the accessibility of our Nextcloud deployment.

Best regards,
Mohamed

**Comment by szaimen** [comment, marked as answer]
Hi Mohamed, 

Nextcloud AIO is in its current form built for community/small scale. For anything beyond that you should get in touch with Nextcloud as a company to ensure Nextcloud performs well on large scale. See https://nextcloud.com/enterprise/

Furthermore note that GitHub is meant for development, so for implementation and bug reporting/features. Community support would be via the community at https://help.nextcloud.com

I hope that makes things clear and have a nice day!

**Reply by melbruki** [reply]
sure no problem, I  plan to have a max of 70 users on the nextcloud aio, instance not sure if that considered large scale. if you could give me a general idea of the setup to accomplish dual stack  this I would appreciate it. otherwise no problem. 

thank you
